### ISSUE TITLE
Independant server loop lag?

### ISSUE THREAD

**Issue by tree786** [lsalzman/enet#175]
Hi

apologies if this is not the ideal place to post, not sure where is best. 
I have created a loop on a server project I am working on, and this loop is outside/independant of the enet poll/receive loop. However I am experiencing unusual lag between the server and client as a result of this, and not sure what I am doing wrong or missing here. Here is a simplified snippet of my server loop (I have cut it down for brevity as original project is more complex):
```
	while(true){
		//SERVER POLL/RECEIVE LOOP
		while(enet_host_service(server,&events,0)>0){
			if (events.type==ENET_EVENT_TYPE_RECEIVE){
				//RECEIVE CLIENT PACKET
				.....
				//DESTROY PACKET
				enet_packet_destroy(event.packet);
			}
		}
		//INDEPENDANT LOOP
		{
		//RUN INDEPENDANT LOGIC HERE
		.....
		//SEND DATA TO CLIENT
		send_packet(packet , event.peer);
		}
	}
```
Previously I had the logic loop inside the poll/receive loop, however that quickly caused issues when more clients connected to the server (as more clients -> more packets received -> faster server updates -> uncontrollable server execution speed). However I did not experience any major lag issues on the client with that setup. I am wondering what the ideal way to handle an independent loop would be (with minimal lag)? Am I using enet incorrectly using this method, or is there a simpler way to achieve this using some enet functionality?

[video showing before change]
[before video.zip](https://github.com/lsalzman/enet/files/7212059/before.video.zip)

[video showing after change]
[after video.zip](https://github.com/lsalzman/enet/files/7212064/after.video.zip)

Thanks

**Comment by lsalzman** [comment]
Make sure that you're sending data to all clients before you go back into the service loop, rather than sending to one client, then servicing, sending to another client, then servicing, etc... So:

while (true) {
  while(service() > 0) { ... }
  for(client : clients) { doSomething(client); }
}

There are further high performance tricks beyond this, but the basics should still work acceptably if done correctly.

**Comment by tree786** [comment]
Thanks , yeah sorry, my original script loops through a vec containing the connected peers and sends to all data to them, rather than to one client. I found that setting a timeout value for `enet_host_service` solved my issue for some reason. I.e. specifying this:
```
int timeout=10;
enet_host_service(server,&events,timeout)>0
```
But when the timeout is 0, i get the slight jitter/lag seen in the video. Of course setting the timeout to a higher value (e.g. 100) blocks the independant loop, so I have kept it lower. Is `enet_host_service` assumed to always take a non-zero value for timeout (i.e. is it bad to specify 0 for it)?
